tools/dependency/pip_check.py

```diff
@@ -11,89 +11,29 @@
+import envoy_repo
 
 
-class PipConfigurationError(Exception):
-    pass
-
-
-class PipChecker(checker.Checker):
-    checks = ("dependabot",)
-    _dependabot_config = DEPENDABOT_CONFIG
-    _requirements_filename = REQUIREMENTS_FILENAME
-
-    @cached_property
-    def config_requirements(self) -> set:
-        """Set of configured pip dependabot directories"""
-        return set(
-            update['directory']
-            for update in self.dependabot_config["updates"]
-            if update["package-ecosystem"] == "pip")
+@abstracts.implementer(pip_check.APipChecker)
+class EnvoyPipChecker:
 
     @cached_property
-    def dependabot_config(self) -> dict:
-        """Parsed dependabot config"""
-        result = utils.from_yaml(self.path.joinpath(self.dependabot_config_path))
-        if not isinstance(result, dict):
-            raise PipConfigurationError(
-                f"Unable to parse dependabot config: {self.dependabot_config_path}")
-        return result
-
-    @property
-    def dependabot_config_path(self) -> str:
-        return self._dependabot_config
-
-    @cached_property
-    def requirements_dirs(self) -> Set[str]:
-        """Set of found directories in the repo containing requirements.txt"""
-        return set(
-            f"/{f.parent.relative_to(self.path)}" for f in self.path.glob("**/*")
-            if f.name == self.requirements_filename)
-
-    @property
-    def requirements_filename(self) -> str:
-        return self._requirements_filename
-
-    def check_dependabot(self) -> None:
-        """Check that dependabot config matches requirements.txt files found in repo"""
-        missing_dirs = self.config_requirements.difference(self.requirements_dirs)
-        missing_config = self.requirements_dirs.difference(self.config_requirements)
-        correct = self.requirements_dirs.intersection(self.config_requirements)
-        if correct:
-            self.dependabot_success(correct)
-        if missing_dirs:
-            self.dependabot_errors(
-                missing_dirs,
-                f"Missing {self.requirements_filename} dir, specified in dependabot config")
-        if missing_config:
-            self.dependabot_errors(
-                missing_config,
-                f"Missing dependabot config for {self.requirements_filename} in dir")
-
-    def dependabot_success(self, correct: Iterable) -> None:
-        self.succeed(
-            "dependabot",
-            ([f"{self.requirements_filename}: {dirname}" for dirname in sorted(correct)]))
-
-    def dependabot_errors(self, missing: Iterable, msg: str) -> None:
-        for dirname in sorted(missing):
-            self.error("dependabot", [f"{msg}: {dirname}"])
+    def path(self) -> pathlib.Path:
+        if self.args.paths:
+            return pathlib.Path(self.args.paths[0])
+        return pathlib.Path(envoy_repo.PATH)
 
 
 def main(*args) -> int:
-    return PipChecker(*args).run()
+    return EnvoyPipChecker(*args).run()
```

Comment:
Are these wrappers just for repository path magic? Is there a way to factor this out? I notice the `path` logic appears multiple times. Otherwise LGTM.

Reply:
i think we want this

if someone has some issue with eg the PipChecker they are going to look at this file and see that the only thing coming locally is the path, and that the rest of the logic is coming from the upstream package

with others it will not just be the `path` - but eg `extension_categories` or `repository_locations` etc etc

the object model here is designed to make these data sources explicit while shifting the actual code upstream